src/main/java/com/microsoft/sqlserver/jdbc/IdleConnectionResiliency.java

```diff
@@ -82,21 +82,24 @@ void resetUnprocessedResponseCount() {
         this.unprocessedResponseCount.set(0);
     }
 
-    void parseInitialSessionStateData(TDSReader tdsReader, byte[][] sessionStateInitial) throws SQLServerException {
+    void parseInitialSessionStateData(byte[] data, byte[][] sessionStateInitial) throws SQLServerException {

```

Comment:
Refactored this to take in the byte[] data instead of the TDSReader. This make calling of it consistent with the other feature ack processing logic.